airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/context/env/ConnectorConfigurationPropertySource.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.context.env
+
+import com.fasterxml.jackson.core.type.TypeReference
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.commons.json.Jsons
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.env.MapPropertySource
+import io.micronaut.core.cli.CommandLine
+import io.micronaut.core.util.StringUtils
+import java.nio.file.Path
+import java.util.AbstractMap
+import java.util.stream.Collectors
+import java.util.stream.Stream
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * Custom Micronaut {@link PropertySource} that reads the command line arguments provided via the
+ * connector CLI and turns them into configuration properties. This allows the arguments to be
+ * injected into code that depends on them via Micronaut. <p /> This property source adds the
+ * following properties to the configuration if the matching value is present in the CLI arguments:
+ * <ul> <li><b>airbyte.connector.operation</b> - the operation argument (e.g. "check", "discover",
+ * etc)</li> <li><b>airbyte.connector.catalog</b> - the Airbyte configured catalog as JSON read from
+ * the configured catalog file path argument, if present.</li> <li><b>airbyte.connector.config</b> -
+ * the normalized Airbyte connector configuration read from the configured connector configuration
+ * file path argument, if present.</li> <li><b>airbyte.connector.state</b> - the Airbyte connector
+ * state as JSON read from the state file path argument, if present.</li> </ol>
+ */
+class ConnectorConfigurationPropertySource(commandLine: CommandLine) :
+    MapPropertySource("connector", resolveValues(commandLine)) {
+    companion object {
+        private const val PREFIX_FORMAT = "%s.%s"
+        private const val ROOT_CONFIGURATION_PROPERTY_KEY = "airbyte.connector"
+        const val CONNECTOR_OPERATION: String = "$ROOT_CONFIGURATION_PROPERTY_KEY.operation"
+        const val CONNECTOR_CONFIG_PREFIX: String = "$ROOT_CONFIGURATION_PROPERTY_KEY.config"
+        const val CONNECTOR_CATALOG_PREFIX: String = "$ROOT_CONFIGURATION_PROPERTY_KEY.catalog"
+        const val CONNECTOR_CATALOG_KEY: String = "configured"
+        const val CONNECTOR_STATE_PREFIX: String = "$ROOT_CONFIGURATION_PROPERTY_KEY.state"
+        const val CONNECTOR_STATE_KEY: String = "json"
```

Comment:
Please improve the names of these keys: `configured` and `json`.

Reply:
Got it. Can you rename the `CONNECTOR_CATALOG_KEY` value from `configured` to `json` then? I guess that tripped me up a bit.

Reply:
Done.